### PR TITLE
Node 12 Upgrade (5-0-x)

### PR DESCRIPTION
**Pull request by nitsakh**
#### Description of Change
This PR uses electrons `electron-node-v12.x` branch which is based on node's v12.x branch.
This is however, not rebased on head of node's v12.x branch yet, but is a few commits behind the v8 7.4 merge in node.
Future PRs will rebase on latest node v12.x commits.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Upgraded to node-12 with v8 7.3

### CODE REVIEW

File: lib/common/asar_init.js [hunk 1]
```diff
@@ -1,14 +1,6 @@
 'use strict'
 
-/* global source, require */
-
-// Expose fs module without asar support.
-// NB: Node's 'fs' and 'internal/fs/streams' have a lazy-loaded circular
-// dependency. So to expose the unmodified Node 'fs' functionality here,
-// we have to copy both 'fs' *and* 'internal/fs/streams' and modify the
-// copies to depend on each other instead of on our asarified 'fs' code.
-source['original-fs'].replace("require('internal/fs/streams')", "require('original-fs/streams')")
```
Comment (zcbenz) [inline lib/common/asar_init.js]: The hack has been moved to our node fork due to changes of node's module loader.

File: atom/browser/node_debugger.cc [hunk 2]
```diff
@@ -38,27 +49,20 @@ void NodeDebugger::Start() {
...
     LOG(ERROR) << "Error parsing node options: "
                << base::JoinString(errors, " ");
   }
 
-  // Set process._debugWaitConnect if --inspect-brk was specified to stop
-  // the debugger on the first line
-  if (options.wait_for_connect()) {
-    mate::Dictionary process(env_->isolate(), env_->process_object());
-    process.Set("_breakFirstLine", true);
-  }
```
[the remark this answers is not in the document]
Comment (zcbenz) [reply atom/browser/node_debugger.cc]: According to the commit message from @nitsakh :

```
    fix: use proper options parser and remove setting of _breakFirstLine
    
    _breakFirstLine was being set on the process, but that has changed in node 12 and so is no longer needed. Node will handle it properly when --inspect-brk is provided
```

We also have a test case for `--inspect-brk`, so it should be fine.

File: atom/browser/node_debugger.cc [hunk 3]
```diff
@@ -17,6 +17,17 @@
 
 #include "atom/common/node_includes.h"
 
+// DebugOptionsParser is defined in node_options.cc
+// We export it from there and then just declare it here for use.
```
[the remark this answers is not in the document]
Comment (zcbenz) [reply atom/browser/node_debugger.cc]: I have refactored the code to export `DebugOptionsParser` from `node_options.h`, so the code here should be clear now.

**Comment by MarshallOfSound** [comment]
What NMV does this currently include?

cc @nornagon

**Comment by nornagon** [comment]
72 https://github.com/electron/node/blob/de2f087468244d2a86e4400803f743c42fa9345a/src/node_version.h#L120

**Comment by nornagon** [comment]
you're totally right, whoops.

**Pull request merged**